cg/meta/store/microsalt.py

```diff
@@ -0,0 +1,182 @@
+            "archive": False,
+            "tag_map_key": tag_map_key,
+        }
+        parsed_files.append(parsed_file)
+
+        if file_["path_index"]:
+            parsed_index_file = {
+                "path": file_["path_index"],
+                "tags": get_tags(
+                    file_, pipeline_tags, analysis_type_tags, tag_map_key, is_index=True
+                ),
+                "archive": False,
+                "tag_map_key": tag_map_key,
+            }
+            parsed_files.append(parsed_index_file)
+
+    return parsed_files
+
+
+def get_tags(
+    file: dict, pipeline_tags: list, tag_map: dict, tag_map_key: tuple, is_index: bool = False
+) -> list:
+    """Get all tags for a file"""
+
+    tags = {
+        "format": file["format"],
+        "id": file["id"],
+        "step": file["step"],
+        "tag": file["tag"],
+    }
+
+    tags["pipeline"] = pipeline_tags[0]
+    tags["application"] = pipeline_tags[1] if len(pipeline_tags) > 1 else None
+
+    tags = _convert_tags(tags, tag_map, tag_map_key, is_index)
+    return tags
+
+
+def _convert_tags(tags: dict, tag_map: dict, tag_map_key: tuple, is_index: bool = False) -> list:
+    """ Convert tags from external deliverables tags to standard internal housekeeper tags """
+
+    if is_index:
+        mapped_tags = tag_map[tag_map_key]["index_tags"]
+    else:
+        mapped_tags = tag_map[tag_map_key]["tags"]
+    converted_tags = [
+        tags["format"],
+        str(tags["id"]),
+        tags["pipeline"],
+    ]
+
+    if tags["application"] is not None:
+        converted_tags.append(tags["application"])
+
+    for tag in mapped_tags:
+        converted_tags.append(tag)
+
+    return sorted(list(set(converted_tags)))
+
+
+def add_new_analysis(bundle_data, microbial_order_obj, status, version_obj):
+    """Function to create and return a new analysis database record"""
+
+    pipeline = HK_TAGS["microsalt"]
+
+    if status.microbial_analysis(
+        microbial_order_id=microbial_order_obj.id, started_at=version_obj.created_at
+    ):
+        raise AnalysisDuplicationError(
+            f"Analysis object already exists for {microbial_order_obj.internal_id} {version_obj.created_at}"
+        )
+
+    new_analysis = status.add_analysis(
+        pipeline=pipeline,
+        started_at=version_obj.created_at,
+        completed_at=dt.datetime.now(),
+        primary=(len(microbial_order_obj.analyses) == 0),
+    )
+    case_obj = status.family(DUMMY_MICROSALT_CASE)
+    new_analysis.family = case_obj
+    new_analysis.microbial_order = microbial_order_obj
+    return new_analysis
+
+
+def _get_microbial_name(deliverables: dict) -> str:
+    """ Get microbial id from deliverables """
+    for file_ in deliverables["files"]:
+        if file_["tag"] == "sampleinfo":
+            return file_["id"]
+
+
+def _get_date_from_results_path(deliverables: dict, project_name: str) -> dt.datetime:
```

Comment:
Is there no other way to get the analysis date from microsalt? Maybe Isak knows?

Reply:
At the moment, no. I did talk to Isak and it's possible to store (and get) the date in some other way, but that will be implemented later as it's a non trivial change. @Isak please correct me if I'm wrong.